src/Morpho.sol

```diff
@@ -484,25 +484,27 @@ contract Morpho is IMorphoStaticTyping {
     /// @dev Assumes that the inputs `marketParams` and `id` match.
     function _accrueInterest(MarketParams memory marketParams, Id id) internal {
         uint256 elapsed = block.timestamp - market[id].lastUpdate;
-
         if (elapsed == 0) return;
 
-        uint256 borrowRate = IIrm(marketParams.irm).borrowRate(marketParams, market[id]);
-        uint256 interest = market[id].totalBorrowAssets.wMulDown(borrowRate.wTaylorCompounded(elapsed));
-        market[id].totalBorrowAssets += interest.toUint128();
-        market[id].totalSupplyAssets += interest.toUint128();
-
-        uint256 feeShares;
-        if (market[id].fee != 0) {
-            uint256 feeAmount = interest.wMulDown(market[id].fee);
-            // The fee amount is subtracted from the total supply in this calculation to compensate for the fact
-            // that total supply is already increased by the full interest (including the fee amount).
-            feeShares = feeAmount.toSharesDown(market[id].totalSupplyAssets - feeAmount, market[id].totalSupplyShares);
-            position[id][feeRecipient].supplyShares += feeShares;
-            market[id].totalSupplyShares += feeShares.toUint128();
-        }
+        if (marketParams.irm != address(0)) {
+            uint256 borrowRate = IIrm(marketParams.irm).borrowRate(marketParams, market[id]);
+            uint256 interest = market[id].totalBorrowAssets.wMulDown(borrowRate.wTaylorCompounded(elapsed));
+            market[id].totalBorrowAssets += interest.toUint128();
+            market[id].totalSupplyAssets += interest.toUint128();
+
+            uint256 feeShares;
+            if (market[id].fee != 0) {
+                uint256 feeAmount = interest.wMulDown(market[id].fee);
+                // The fee amount is subtracted from the total supply in this calculation to compensate for the fact
+                // that total supply is already increased by the full interest (including the fee amount).
+                feeShares =
+                    feeAmount.toSharesDown(market[id].totalSupplyAssets - feeAmount, market[id].totalSupplyShares);
+                position[id][feeRecipient].supplyShares += feeShares;
+                market[id].totalSupplyShares += feeShares.toUint128();
+            }
 
-        emit EventsLib.AccrueInterest(id, borrowRate, interest, feeShares);
+            emit EventsLib.AccrueInterest(id, borrowRate, interest, feeShares);
+        }
 
         // Safe "unchecked" cast.
         market[id].lastUpdate = uint128(block.timestamp);
```

Comment:
What's the "meaning" of `lastUpdate`? I'm asking because there's no natspec documentation on the interface, and previously it was updated when there was an interest accrual.

So if I need to give a meaning from the previous code logic, `lastUpdate` represented the last time that interest was accrued or at least triggered (maybe there was no accrual because of market condition or rounding)

With this change instead, `lastUpdate` is always updated even if no interest accrual is triggered, and we know for sure that the market has not changed.

Should the `market[id].lastUpdate = uint128(block.timestamp);` execution be moved inside the `if` branch?

Reply:
I think now it refers to the timestamp of the block where there was the last interaction that could change the interest. So mostly the main lending entry points except `supplyCollateral`. I agree that it does not really make the spec simpler in the end, so maybe it's better to use the previous solution of skipping the all function `_accrueInterest` as soon as the irm is 0

Reply:
Do you mean something like `if (elapsed == 0 || marketParams.irm == address(0)) return;`?

Reply:
We thought about returning even earlier, before computing `elapsed`, so that we would not have to touch the fee/lastUpdate slot. But this is not such a big optimization actually because this slot is warm anyway (when checking if the market is created). So we thought about your suggestion also.

In the end, because optimizations are not so big, we went for the solution where markets with 0 irm function similarly to regular markets

Reply:
I think that lastUpdate should always be updated (current situation) because it makes the spec simpler (it's always updated when accrueInterest is called). Also I was in favor (but not big deal) of always emitting the event for the same reason. For me when irm=0 we can say that the interest is accrued, but with rate=0.